src/compiler/checker.ts

```diff
@@ -8409,6 +8449,86 @@ namespace ts {
             return links.resolvedType;
         }
 
+        function getTypeFromImportTypeNode(node: ImportTypeNode): Type {
+            const links = getNodeLinks(node);
+            if (!links.resolvedType) {
+                if (node.isTypeOf && node.typeArguments) { // Only the non-typeof form can make use of type arguments
+                    error(node, Diagnostics.type_arguments_can_only_be_used_in_a_ts_file);

```

Comment:
that error message does not seem right, you probably want something like `typeof can not b sued on a type` or `type arguments can not be used on a value` or just `type arguments can not be used here`

Reply:
'Doh, I put this one here as a placeholder and forgot to replace it with a new message. Thanks for the catch!